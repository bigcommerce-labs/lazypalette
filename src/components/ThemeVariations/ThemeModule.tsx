import React from 'react';

import { Item, List, Thumb, Title } from './styles';
import { ThemePropsList } from './ThemeVariations';

const ThemeModule = (props: {variants: ThemePropsList}) => (
    <List>
        {props.variants.map(({name, image, isActive}) => (
            <Item key={name} isActive={isActive}>
                <Title>{name}</Title>
                <Thumb previewPath={image} />
            </Item>
        ))}
    </List>
);

export default ThemeModule;
