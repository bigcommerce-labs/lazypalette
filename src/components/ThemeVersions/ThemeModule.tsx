import React from 'react';

import { ThemePropsList } from './ThemeVersions';
import { Header, Item, List, MenuModal, NavItem, Thumb, Title, Wrapper } from './styles';

const ThemeModule = (props: { variants: ThemePropsList }) => (
  <Wrapper>
    <MenuModal>
      <Header>Store theme</Header>
      <NavItem to="/design" />
      <List>
        {props.variants.map(variant => (
          <Item key={variant.name}>
            <Title>{variant.name}</Title>
            <Thumb theme={variant.image} />
          </Item>
        ))}
      </List>
    </MenuModal>
  </Wrapper>
);

export default ThemeModule;
