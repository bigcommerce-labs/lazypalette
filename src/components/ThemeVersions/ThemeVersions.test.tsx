import { shallow } from 'enzyme';
import React from 'react';
import ThemeModule from './ThemeModule';

it('renders', () => {
  const testItems = [
    {
      image: 'http://meow.wow.com/123.jpg',
      name: 'light',
    },
    {
      image: 'http://meow.wow.com/123.jpg',
      name: 'dark',
    },
  ];

  const themeModule = shallow(
    <ThemeModule variants={testItems} />
  );

  expect(themeModule).toMatchSnapshot();
});
