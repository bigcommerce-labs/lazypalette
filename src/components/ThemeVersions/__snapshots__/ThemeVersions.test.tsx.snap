// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<styled.div>
  <styled.div>
    <styled.h2>
      Store theme
    </styled.h2>
    <Styled(Link)
      to="/design"
    />
    <styled.ul>
      <styled.li
        key="light"
      >
        <styled.div>
          light
        </styled.div>
        <styled.div
          theme="http://meow.wow.com/123.jpg"
        />
      </styled.li>
      <styled.li
        key="dark"
      >
        <styled.div>
          dark
        </styled.div>
        <styled.div
          theme="http://meow.wow.com/123.jpg"
        />
      </styled.li>
    </styled.ul>
  </styled.div>
</styled.div>
`;
